snaps/openvino/gpu-support/snap/snapcraft.yaml

```diff
@@ -0,0 +1,145 @@
+name: openvino-python
+base: core24
+version: '2024.4.0'
+summary: Snap for testing the OpenVINO Python API
+description: |
+  OpenVINO is a software toolkit for optimizing and deploying deep learning models.
+  This snap includes support for GPU and NPU acceleration, and ships ipython and
+  jupyter for ease of use.
+grade: devel
+confinement: strict
+
+plugs:
+  intel-npu:
+    interface: custom-device
+    custom-device: intel-npu-device
+  npu-libs:
+    interface: content
+    content: npu-libs-2404
+    target: $SNAP/usr/lib/npu-libs-2404
+    #default-provider: intel-npu-driver # must be in latest/stable channel so commenting for now
+  openvino-libs:
+    interface: content
+    content: openvino-toolkit-2404
+    target: $SNAP/openvino
+    #default-provider: openvino-toolkit-2404 # must be in latest/stable channel so commenting for now
+
+apps:
+
+  ipython:
+    command: bin/ipython
+    plugs:
+      - intel-npu # NPU device
+      - npu-libs
+      - openvino-libs
+      - network # for common tasks like downloading datasets
+      - opengl # GPU device
+      - home
+      - hugepages-control
+    environment:
+      PYTHONPATH: $SNAP/usr/lib/python3/dist-packages:$SNAP/openvino/usr/python:$SNAP/openvino/usr/python/openvino/tools/ovc:$PYTHONPATH
+      LD_LIBRARY_PATH: $SNAP/openvino/usr/runtime/lib/intel64:$SNAP/openvino/usr/lib/x86_64-linux-gnu:$SNAP/usr/lib/npu-libs-2404:$SNAP/usr/local/lib:$LD_LIBRARY_PATH
+      PATH: $SNAP/openvino/usr/python/openvino/tools/ovc:$PATH
+      OCL_ICD_VENDORS: $SNAP/etc/OpenCL/vendors
+
+  jupyter:
+    command: bin/jupyter
+    plugs:
+      - intel-npu # GPU device
+      - npu-libs
+      - openvino-libs
+      - network # for common tasks like downloading datasets
+      - opengl # GPU device
+      - home
+      - network-bind # needed by jupyter
+      - hugepages-control
+    environment:
+      PYTHONPATH: $SNAP/usr/lib/python3/dist-packages:$SNAP/openvino/usr/python:$SNAP/openvino/usr/python/openvino/tools/ovc:$PYTHONPATH
+      LD_LIBRARY_PATH: $SNAP/openvino/usr/runtime/lib/intel64:$SNAP/openvino/usr/lib/x86_64-linux-gnu:$SNAP/usr/lib/npu-libs-2404:$SNAP/usr/local/lib:$LD_LIBRARY_PATH
+      PATH: $SNAP/openvino/usr/python/openvino/tools/ovc:$PATH
+      OCL_ICD_VENDORS: $SNAP/etc/OpenCL/vendors
+
+parts:
+  openvino-python-deps:
+    # This part will install a bunch of python tools and libraries that
+    # are needed to run and validate the GPU device notebook at:
+    # https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/gpu-device
+    source-type: git
+    source: https://github.com/openvinotoolkit/openvino.git
```

Comment:
is it useful to align the version we use here with the openvino toolkit by adding `source-tag: 2024.4.0` ?
i understand that this snap is only a sample snap (a reference implementation) however for the long run, i m wondering how we can enforce the compatibility between this kind of consumer snap wrt to `openvino-toolkit` producer snap

Reply:
Yes, for sure, that was a miss on my part! Good catch.